### PR TITLE
Fix `UNICODE` conflict on Windows

### LINKED ISSUE #27568: Windows nightly build errors
## 🐛 Bug

There are build errors with https://download.pytorch.org/libtorch/nightly/cu100/libtorch-win-shared-with-deps-latest.zip
1. Unicode build
2. compile issue

1. Set Unicode carachter set in the project settings: you have an error .\libtorch\include\torch\csrc\jit\pickler.h line 34 with 
`UNICODE` = 'V',

2. In \libtorch\include\c10\util\llvmMathExtras.h
```
inline int64_t minIntN(int64_t N) {
   assert(N > 0 && N <= 64 && "integer width out of range");

   return -(UINT64_C(1)<<(N-1));
 }
```
does not compile, should be changed in 
```
 inline int64_t minIntN(int64_t N) {
   assert(N > 0 && N <= 64 && "integer width out of range");
   int64_t n = (UINT64_C(1) << (N - 1));
   return -n;
 }
```

cc @peterjc123

### PR DESCRIPTION
**Pull request by peterjc123**
Fixes https://github.com/pytorch/pytorch/issues/27568.
cc @IlyaOvodov.

### CODE REVIEW

**Comment by ezyang** [comment]
Hmm... I would have just renamed the opcode. What do you think? (I get especially nervous about mucking around with macros in header files)

File: torch/csrc/jit/pickler.h [hunk 1]
```diff
@@ -11,6 +11,13 @@
 namespace torch {
 namespace jit {
 
+// In order to solve the conflict between the macro and the variable in MSVC,
+// we need to save the value first to `UNICODE_TMP` and undefine `UNICODE`.
+#ifdef UNICODE
+# define UNICODE_TMP UNICODE
+# undef UNICODE
+#endif
+
```
Comment (peterjc123) [inline torch/csrc/jit/pickler.h]: @ezyang You mean the one in L41? Could you please suggest a good one?

Comment (soumith) [reply torch/csrc/jit/pickler.h]: `TORCH_UNICODE`?

Comment (ezyang) [reply torch/csrc/jit/pickler.h]: or maybe `UNICODE_` (trailing underscore being a classic way of sidestepping reserved words)

**Pull request closed without merging**